### PR TITLE
Further ci tweaks

### PR DESCRIPTION
**Pull request by jshjohnson**

### CODE REVIEW

File: .github/workflows/unit-tests.yml [hunk 1]
```diff
@@ -33,8 +33,7 @@ jobs:
         run: bash <(curl -s https://codecov.io/bash)
           -f ./coverage/lcov.info
           -B ${{ github.head_ref }}
-          -C ${{ github.event.after }}
-          -P ${{ github.event.number }}
+          -C ${{ github.sha }}
```
Comment (tinovyatkin) [inline .github/workflows/unit-tests.yml]: `P` here was the Pull Request number, it helps to have backlinks at Codecov dashboard to a specific request

Comment (jshjohnson) [reply .github/workflows/unit-tests.yml]: Ah okay - it seemed to be having issues with the value passed to `-C`. I couldn't find where these arguments are documented https://docs.codecov.io/docs/about-the-codecov-bash-uploader#section-flag

Comment (jshjohnson) [reply .github/workflows/unit-tests.yml]: `==> GitLab CI detected.
    project root: .
    Yaml found at: .codecov.yml
    -> Found 1 reports
==> Detecting git/mercurial file structure
==> Appending build variables
    + github-action
==> Reading reports
    + ./coverage/lcov.info bytes=43031
==> Appending adjustments
    http://docs.codecov.io/docs/fixing-reports
    + Found adjustments
==> Gzipping contents
==> Uploading reports
    url: https://codecov.io
    query: branch=increase-unit-test-coverage&commit=-P&build=&build_url=&name=&tag=&slug=jshjohnson%2FChoices&service=gitlab&flags=&pr=&job=
    -> Pinging Codecov
https://codecov.io/upload/v4?package=bash-8a28df4&token=secret&branch=increase-unit-test-coverage&commit=-P&build=&build_url=&name=&tag=&slug=jshjohnson%2FChoices&service=gitlab&flags=&pr=&job=
    -> Uploading
    X> Failed to upload
`

Comment (tinovyatkin) [reply .github/workflows/unit-tests.yml]: > I couldn't find where these arguments are documented https://docs.codecov.io/docs/about-the-codecov-bash-uploader#section-flag

Not exactly documented, but https://github.com/codecov/codecov-bash/blob/1044b7a243e0ea0c05ed43c2acd8b7bb7cef340c/codecov#L158

Comment (tinovyatkin) [reply .github/workflows/unit-tests.yml]: > ```
> https://codecov.io/upload/v4?package=bash-8a28df4&token=secret&branch=increase-unit-test-coverage&commit=-P&build=&build_url=&name=&tag=&slug=jshjohnson%2FChoices&service=gitlab&flags=&pr=&job=
>     -> Uploading
>     X> Failed to upload
>     -> Sleeping for 30s and trying again...
>     -> Uploading to Codecov
>     HTTP 400
> commit must match pattern ^\d+:\w{12}|\w{40}$
> ```

`github.event.after` exists while Action is running on a `pull_request` trigger, and it's not the same as `github.sha` in that case (it refers to the latest commit).
That's why on `master` push run I'm using `github.sha` for commit value.

**Pull request merged**